features/FEATURE_BLE/source/generic/GenericGap.cpp

```diff
@@ -570,25 +570,21 @@ ble_error_t GenericGap::stopScan()
 {
     ble_error_t err;
 
-    if (is_extended_advertising_available()) {
-        if (!_scan_enabled) {
-            return BLE_ERROR_NONE;
-        }
+    if (!_scan_enabled) {
+        return BLE_ERROR_NONE;
+    }
 
-        _scan_enabled = false;
+    _scan_enabled = false;
 
+    if (is_extended_advertising_available()) {
         err = _pal_gap.extended_scan_enable(false, pal::duplicates_filter_t::DISABLE, 0, 0);
-
-        if (err) {
-            _scan_enabled = true;
-            return err;
-        }
     } else {
         err = _pal_gap.scan_enable(false, false);
+    }
```

Comment:
I believe we should stop the timeout as well.

Reply:
already detached